fem/nonlininteg.hpp

```diff
@@ -285,6 +330,42 @@ class IncompressibleNeoHookeanIntegrator : public BlockNonlinearFormIntegrator
                                     const Array2D<DenseMatrix *> &elmats);
 };
 
+class VectorConvectionNLFIntegrator : public NonlinearFormIntegrator
+{
+private:
+   Coefficient *Q{};
+   DenseMatrix dshape, dshapex, EF, gradEF, ELV, elmat_comp;
+   Vector shape;
+   // PA extension
+   Vector pa_data;
+   const DofToQuad *maps;         ///< Not owned
+   const GeometricFactors *geom;  ///< Not owned
+   int dim, ne, nq;
+public:
+   VectorConvectionNLFIntegrator(Coefficient &q): Q(&q) { }
+
+   VectorConvectionNLFIntegrator() = default;
+
+   static const IntegrationRule &GetRule(const FiniteElement &fe,
+                                         ElementTransformation &T);
+
+   virtual void AssembleElementVector(const FiniteElement &el,
+                                      ElementTransformation &trans,
+                                      const Vector &elfun,
+                                      Vector &elvect);
+
+   virtual void AssembleElementGrad(const FiniteElement &el,
+                                    ElementTransformation &trans,
+                                    const Vector &elfun,
+                                    DenseMatrix &elmat);
+
+   using NonlinearFormIntegrator::AssemblePA;
+
+   virtual void AssemblePA(const FiniteElementSpace &fes);
+
+   virtual void MultPA(const Vector &x, Vector &y) const;
```

Comment:
This class should implement `AddMultPA` -- remove `MultPA`.

Reply:
Done.